File: Week 1/Problem Set 1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    int C;
    do
    {
        C = get_int("Change owed: ");
    }
    while (C < 0);
    return C;
}

int calculate_quarters(int cents)
{
    int Q = 0;
    while (cents >= 25)
    {
        Q++;
        cents = cents - 25;
    }
    return Q;
}

int calculate_dimes(int cents)
{
    int D = 0;
    while (cents >= 10)
    {
        D++;
        cents = cents - 10;
    }
    return D;
}

int calculate_nickels(int cents)
{
    int N = 0;
    while (cents >= 5)
    {
        N++;
        cents = cents - 5;
    }
    return N;
}

int calculate_pennies(int cents)
{
    int P = 0;
    while (cents >= 1)
    {
        P++;
        cents = cents - 1;
    }
    return P;
}

File: Week 6/Problem Set 6/sentimental-cash/cash.py
# TODO
from cs50 import get_float

# Getting user input
while True:
    change = get_float("Change owed: ")
    if change > 0:
        break

# Calculating the minimum number of coins
min = 0

# Quarters
while True:
    if change >= 0.25:
        change = round(change - 0.25, 2)
        min += 1
    else:
        break

# Dimes
while True:
    if change >= 0.10:
        change = round(change - 0.10, 2)
        min += 1
    else:
        break

# Nickels
while True:
    if change >= 0.05:
        change = round(change - 0.05, 2)
        min += 1
    else:
        break

# Pennies
while True:
    if change >= 0.01:
        change = round(change - 0.01, 2)
        min += 1
    else:
        break

# Printing the value
print(f"{min}")
